src/schematools/contrib/django/management/commands/import_schemas.py

```diff
@@ -62,16 +63,50 @@ def import_from_url(self, schema_url) -> List[Dataset]:
 
     def import_schema(self, name: str, schema: DatasetSchema) -> Optional[Dataset]:
         """Import a single dataset schema."""
+
+        created = False
         try:
-            dataset = Dataset.objects.get(name=to_snake_case(schema.id))
+            dataset = Dataset.objects.get(name=Dataset.name_from_schema(schema))
         except Dataset.DoesNotExist:
-            dataset = Dataset.create_for_schema(schema)
+            try:
+                # try getting default dataset by name and version
+                dataset = Dataset.objects.filter(Q(version=None) | Q(version=schema.version)).get(
+                    name=to_snake_case(schema.id)
```

Comment:
I guess this should also use `Dataset.name_from_schema()`, so better have it defined once in the method.